src/modules/hip/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,1439 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+#include "reduction.hpp"
+
+// -------------------- Set 0 - Reduction Stage 2 --------------------
+__global__ void tensor_sum_grid_result(Rpp32u *srcPtr,
+                                       uint xBufferLength,
+                                       Rpp64u *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ uint partialSumShared[1024];                         // 8192 uints of src reduced to 1024 in a 1024 x 1 thread block
```

Comment:
Rename all instances of <>Shared to <>_smem as per new agreement for now

Reply:
sure